src/libtools/calculator.cc

```diff
@@ -38,39 +35,31 @@ void Calculator::UpdateWithDefaults(Property &options, std::string package) {
 
   Property defaults_all;
   defaults_all.LoadFromXML(xmlFile);
-  Property defaults = defaults_all.get("options." + id);
+  _options = defaults_all.get("options." + id);
+}
 
-  // if a value not given or a tag not present, provide default values
-  AddDefaults(options_id, defaults);
+void Calculator::UpdateWithUserOptions(Property &options) {
 
-  // output calculator options
-  std::string indent("          ");
-  Index level = 1;
-  votca::tools::PropertyIOManipulator IndentedText(PropertyIOManipulator::TXT,
-                                                   level, indent);
-  if (Log::verbose()) {
-    std::cout << "\n... ... options\n"
-              << IndentedText << options_id << "... ... options\n"
-              << std::flush;
-  }
+  // copy options from the object supplied by the Application
+  std::string id = Identify();
+  Property options_id = options.get("options." + id);
+
+  // if a value is given override default values
+  AddDefaults(options_id, _options);
 }
 
-void Calculator::AddDefaults(Property &p, const Property &defaults) {
+void Calculator::AddDefaults(Property &p, Property &defaults) {
```

Comment:
why not const?

Reply:
I have added then it the const in the next commits